src/OpenApi/test/Services/OpenApiDocumentServiceTests.Paths.cs

```diff
@@ -0,0 +1,158 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.OpenApi.Models;
+
+public partial class OpenApiDocumentServiceTests : OpenApiDocumentServiceTestBase
+{
+    [Fact]
+    public async Task GetOpenApiPaths_ReturnsPaths()
+    {
+        // Arrange
+        var builder = CreateBuilder();
+
+        // Act
+        builder.MapGet("/api/todos", () => { });
+        builder.MapGet("/api/users", () => { });
+
+        // Assert
+        await VerifyOpenApiDocument(builder, document =>
+        {
+            Assert.Collection(document.Paths.OrderBy(p => p.Key),
+                path =>
+                {
+                    Assert.Equal("/api/todos", path.Key);
+                    Assert.Collection(path.Value.Operations.OrderBy(o => o.Key),
+                        operation =>
+                        {
+                            Assert.Equal(OperationType.Get, operation.Key);
+                        });
+                },
+                path =>
+                {
+                    Assert.Equal("/api/users", path.Key);
+                    Assert.Collection(path.Value.Operations.OrderBy(o => o.Key),
+                        operation =>
+                        {
+                            Assert.Equal(OperationType.Get, operation.Key);
+                        });
+                });
+        });
+    }
+
+    [Fact]
+    public async Task GetOpenApiPaths_RespectsShouldInclude()
```

Comment:
Is this using the default `ShouldInclude`?  Do I recall correctly that that predicate will force the group name to match the (default) document name, v1?

Reply:
> Is this using the default ShouldInclude? 

Yep.

> Do I recall correctly that that predicate will force the group name to match the (default) document name, v1?

Correct, hence the assertion that only the endpoint with the `v1` group name ends up being populated in this document.

Reply:
That's a lot of context you need to have to read this test.  Some comments might help.

Reply:
Popped a comment into the assertion to clarify what is going on here.